### PR TITLE
extendCli Plugin Option API

### PR DESCRIPTION
**Pull request by ulivz**
**Summary**

It provides a better way to fulfill these requirements of plugin logic outside of `dev` and `build`.

e.g.

- export site to a pdf file, #919 #546
- custom a `info` command to display the site info in command line.

Internal changes:

1. Removed '@vuepress/cli' package;
2. Modified the compatible node version to 8.6 or higher. 

**What kind of change does this PR introduce?** (check at least one)

- [ ] Bugfix
- [x] Feature
- [ ] Code style update
- [x] Refactor
- [x] Docs
- [ ] Build-related changes
- [ ] Other, please describe:

If changing the UI of default theme, please provide the **before/after** screenshot:

**Does this PR introduce a breaking change?** (check one)

- [ ] Yes
- [x] No

If yes, please describe the impact and migration path for existing applications:

**The PR fulfills these requirements:**

- [ ] When resolving a specific issue, it's referenced in the PR's title (e.g. `fix #xxx[,#xxx]`, where "xxx" is the issue number)

You have tested in the following browsers: (Providing a detailed version will be better.)

- [x] Chrome
- [ ] Firefox
- [x] Safari
- [ ] Edge
- [ ] IE

If adding a **new feature**, the PR's description includes:

- [x] A convincing reason for adding this feature
- [ ] Related documents have been updated
- [ ] Related tests have been updated

To avoid wasting your time, it's best to open a **feature request issue** first and wait for approval before working on it.

**Other information:**

// @egoist @sulmanweb @LinFeng1997

### CODE REVIEW

File: packages/docs/docs/plugin/option-api.md [hunk 1]
```diff
@@ -434,3 +434,29 @@ Then, VuePress will automatically inject these components behind the layout comp
...
+
+- Type: `function`
+- Default: `undefined`
+
+Register a extra command to enhance the CLI of vuepress. The function will be called with a [CAC](https://github.com/cacjs/cac)'s instance as the first argument.
+
+```js
+module.exports = {
+  registerCommand (cli) {
+    cli
```
Comment (egoist) [inline packages/docs/docs/plugin/option-api.md]: why call it `registerCommand` when you can get the entire `cli` instance 😂

Comment (ulivz) [reply packages/docs/docs/plugin/option-api.md]: How is `extendCLI`?

Comment (ulivz) [reply packages/docs/docs/plugin/option-api.md]: Maybe passing a proxied CLI instance would be better, for now we should only have the ability ti extend the command.

Comment (ulivz) [reply packages/docs/docs/plugin/option-api.md]: Renamed to `extendCli` at https://github.com/vuejs/vuepress/pull/1069/commits/1f7bf85bdd7e96e4f759d3785c96b60e32e5fa83

**Pull request merged**